commit e5dae9f3731753dc5ef57e081a7b45efd71e7545 (refs/changes/55/555/3)
Author: Cody Peter Mello <cody.mello@joyent.com>
Date:   2016-10-10T18:36:12+00:00 (3 years ago)
    
    OS-5687 Update to GNU screen 4.4.0-32
    Reviewed by: Brian Bennett <brian.bennett@joyent.com>
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Approved by: Patrick Mooney <patrick.mooney@joyent.com>

diff --git a/screen/Makefile b/screen/Makefile
index 27b247b..26d82ed 100644
--- a/screen/Makefile
+++ b/screen/Makefile
@@ -18,15 +18,22 @@
 #
 # CDDL HEADER END
 #
-# Copyright (c) 2012, Joyent, Inc.  All rights reserved.
+# Copyright 2016, Joyent, Inc.  All rights reserved.
 #
 
-VER =	screen-4.0.3
+VER =	screen-4.4.0
 
 include ../Makefile.defs
 
 PATCHES =	Patches/*
 
+#
+# We set _XOPEN_SOURCE=500 so that screen can access additional fields in
+# struct msghdr. pkgsrc also does this.
+#
+AUTOCONF_ENV += \
+	CFLAGS=-D_XOPEN_SOURCE=500
+
 #
 # It is important to make sure that screen has a prefix of /usr. When screen
 # builds and prefix is not set, it runs which screen to determine the prefix. On
diff --git a/screen/Patches/misc.c.patch b/screen/Patches/misc.c.patch
deleted file mode 100644
index 738f990..0000000
--- a/screen/Patches/misc.c.patch
+++ /dev/null
@@ -1,12 +0,0 @@
---- screen-4.0.3/misc.c	2003-12-05 19:15:41.000000000 +0530
-+++ screen-4.0.3/misc.c	2006-04-03 16:46:19.150133000 +0530
-@@ -613,7 +613,7 @@
-    */
- # endif /* NEEDSETENV */
- #else /* USESETENV */
--# if defined(linux) || defined(__convex__) || (BSD >= 199103)
-+# if defined(linux) || defined(__convex__) || (BSD >= 199103) || (__sun)
-   setenv(var, value, 1);
- # else
-   setenv(var, value);
-
diff --git a/screen/Patches/screen.1.patch b/screen/Patches/screen.1.patch
index 848f2e4..d0252f9 100644
--- a/screen/Patches/screen.1.patch
+++ b/screen/Patches/screen.1.patch
@@ -2,7 +2,7 @@ diff --git a/doc/screen.1 b/doc/screen.1
 index 348d357..6886d2a 100644
 --- a/doc/screen.1
 +++ b/doc/screen.1
-@@ -594,7 +594,7 @@ in the environment variable $SCREENDIR.
+@@ -649,7 +649,7 @@ in the environment variable $SCREENDIR.
  When
  .I screen
  is invoked, it executes initialization commands from the files
@@ -11,7 +11,7 @@ index 348d357..6886d2a 100644
  \*Q.screenrc\*U in the user's home directory. These are the \*Qprogrammer's
  defaults\*U that can be overridden in the following ways: for the
  global screenrc file 
-@@ -4416,12 +4416,12 @@ Examples in the
+@@ -4903,12 +4903,12 @@ Examples in the
  .I screen
  distribution package for private and global initialization files.
  .IP $SYSSCREENRC 
@@ -23,6 +23,6 @@ index 348d357..6886d2a 100644
  .IP $HOME/.screenrc
 -Read in after /usr/local/etc/screenrc
 +Read in after /etc/screenrc
- .IP $SCREENDIR/S-<login>
- .IP /local/screens/S-<login>
+ .IP $SCREENDIR/S\-<login>
+ .IP /local/screens/S\-<login>
  Socket directories (default)
diff --git a/screen/screen-4.0.3.tar.gz b/screen/screen-4.0.3.tar.gz
deleted file mode 100644
index 860819d..0000000
Binary files a/screen/screen-4.0.3.tar.gz and /dev/null differ
diff --git a/screen/screen-4.4.0.tar.gz b/screen/screen-4.4.0.tar.gz
new file mode 100644
index 0000000..2d9f50d
Binary files /dev/null and b/screen/screen-4.4.0.tar.gz differ
